proc_IR.py

```diff
@@ -1,77 +1,273 @@
+    fl.plot(s_sliced.imag,'o')
+    #}}}
+
+    s.ift('t2')
+    fl.next('t domain for apod')
+    fl.image(s.C.setaxis('vd','#').set_units('vd','scan #'))
+    #{{{time shift and apodization
+    t0 = s['ph2',ph2_val]['ph1',ph1_val].C.mean_all_but('t2').argmax('t2').item()
+    s.setaxis('t2', lambda x: x - t0)
+    apod = s.fromaxis('t2') * 0
+    apod['t2':(None,30e-3)] = apod['t2':(None,30e-3)].run(lambda x: tukey(len(x)))
+    s *= apod
     #}}}
-    #{{{recovery curve and fitting
-    s_sliced = s['ph2',coh_sel['ph2']]['ph1',coh_sel['ph1']]*-1 # bc inverted at high powers
-    # below, f_range needs to be defined
-    M0,Mi,R1,vd = sympy.symbols("M_0 M_inf R_1 indirect",real=True)
-    f,T1,g = recovery(s_sliced, (-100,100),
-            guess={M0:-500, Mi:500, R1:1})
-    fl.plot_curve(f,'inversion recovery curve',guess=g)
-fl.show()
+    if postproc == 'spincore_IR_v1':
+        #{{{subplot for imaging before alignment
+        s.ft('t2')
+        s.reorder('ph1',first=True)
+        s.reorder('ph2',first=True)
+        s.reorder('t2',first=False)
+        fl.next('freq domain - before corr, overview')
+        fl.image(s.C.setaxis(
+    'vd','#').set_units('vd','scan #'))
+        fig,(ax1,ax2) = plt.subplots(1,2)
+        fl.next('filtered and apodized -- before alignment', fig=fig)
+        fl.image(s['ph2',ph2_val]['ph1',ph1_val].C.setaxis(
+    'vd','#').set_units('vd','scan #'),ax=ax1)
+        fl.image(abs(s)['ph2',ph2_val]['ph1',ph1_val].C.setaxis(
+    'vd','#').set_units('vd','scan #'),ax=ax2)
+        #}}}
+        #{{{ pre-alignment
+        s_final = s.C
+        s_before = s['vd',:2]
+        s_after = s['vd',3:]
+        s_after.ift(['ph2','ph1'])
+        s_after.ift('t2')
+        s_after.ft('t2')
+        fl.next('before alignment - FT')
+        fl.image(s_after.C.setaxis(
+    'vd','#').set_units('vd','scan #'), interpolation='bilinear')
+        s_before.ift(['ph2','ph1'])
+        s_before.ift('t2')
+        s_before.ft('t2')
+        fl.next('prior to alignment')
+        fl.image(s_before, interpolation = 'bilinear')
+        #}}}
+        #{{{alignment for before and after null
+        fl.basename='first pass'
+        opt_shift_before,sigma_before = correl_align(s_before,indirect_dim='vd',
+                ph1_selection=ph1_val,ph2_selection=ph2_val,fl=fl)
+        fl.basename = None
+        fl.basename ='first pass after null'
+        opt_shift_after,sigma_after = correl_align(s_after,indirect_dim='vd',
+                ph1_selection=ph1_val,ph2_selection=ph2_val,fl=fl)
+        fl.basename = None
+        s_before.ift('t2')
+        s_before *= np.exp(-1j*2*pi*opt_shift_before*s_before.fromaxis('t2'))
+        fl.next('out of correl alignment before')
+        fl.image(s_before)
+        s_after.ift('t2')
+        s_after *= np.exp(-1j*2*pi*opt_shift_after*s_after.fromaxis('t2'))
+        fl.next('out of correl alignment after')
+        fl.image(s_after.C.setaxis(
+    'vd','#').set_units('vd','scan #'))
```

Comment:
this is not the proper way to indent a continued line, and I actually don't understand how this happens.  If your using gvim, vscode, etc, it would not naturally do this.

If using gvim, fix by highlighting and pressing `=`.  I am fixing this in several places rather than commenting on each one.